### ISSUE TITLE
Issue running define-env with a value that has a space on CI

### ISSUE THREAD

**Issue by sbatezat** [ibrahim-mubarak/define-env#11]
On CI (docker unix image but I'm able to reproduce on windows Powershell), I'm using this tool like that:

`flutter build web $(flutter pub run define_env -f "env/.staging.env")`

Everything is working fine if .staging.env file doesn't contains space but it's failing each time there is one (KEY="VALUE WITH SPACE" for example)

It's exactly the same than that issue: https://github.com/flutter/flutter-intellij/issues/4588. After some investigation, I think I understand why, but can't figure out how to fix it. The output of the command in interpreted as a string, and the flutter build/run command is not working fine in these cases.

Relevant working samples:

`flutter build web --dart-define=KEY1=VALUE1 --dart-define=KEY2="VALUE2 WITH SPACE"`
`flutter build web '--dart-define=KEY2="VALUE2 WITH SPACE"'`
`flutter build web '--dart-define=KEY1=VALUE1' '--dart-define=KEY2="VALUE2 WITH SPACE"'`

Relevant NOT working samples:

`flutter build web '--dart-define=KEY1=VALUE1 --dart-define=KEY2="VALUE2 WITH SPACE"'`
=> output "Target file "WITH" not found."

The thing is I'm not sure I can raise an issue to flutter team as the command line argument is not designed to be inserted between quotes. I think the real issue is on the way "define_env" is outputing the result.

**Comment by sbatezat** [comment]
It seems there is now a native way to "define-env" from file (https://github.com/flutter/flutter/issues/107810).
I will have a look at this feature and may switch to it!

**Comment by ibrahim-mubarak** [comment]
So it looks like the variables not being expanded properly
Can you try the following

```bash
flutter pub run define_env -f "env/.staging.env" | xargs flutter build web
```

Also thanks for pointing out the native way for doing this.
I will reevaluate how `define_env` can be used once lands in stable, which might take another month or so.

**Comment by sbatezat** [comment]
> 

I'm now using "--dart-define-from-file" with JSON file (available since Flutter 3.7.0) and I'm no longer using define-env.
Thanks anyway, it was very helpful!
